src/inet/InetInterface.cpp

```diff
@@ -226,6 +226,125 @@ void CloseIOCTLSocket(void)
 
 #if WEAVE_SYSTEM_CONFIG_USE_SOCKETS
 
+#if __ANDROID_API__ >= 21 && __ANDROID_API__ < 24
```

Comment:
Could this be:

`#if __ANDROID__ && __ANDROID_API__ < 24` instead? That way this can also work when building for < 21, as well as for non-Android targets.

Reply:
I think we should wait until we understand the utility of this work-around before we apply it to non-Android targets.

Reply:
Sorry, to clarify I just meant that having the __ANDROID__ flag would ensure it's *only* for Android, but applies to Android for all API levels < 24.

Reply:
Agreed.  I believe we only use `__ANDROID__API__ > = 21`, until we have a use case here, I'd hold off.

Reply:
Currently the PairingKit library in google3 compiles with 19, so that means it will fail to compile, as it will be using the API 24 function. There should be no need to specify a minimum here, just that it's android and < 24.

Reply:
OK, will change accordingly

Reply:
changed